src/Model/FileSupport/BuyRequest/CustomizableOptionDataProvider.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/quote-graphql
+ * @link    https://github.com/scandipwa/quote-graphql
+ */
+
+declare(strict_types=1);
+
+namespace ScandiPWA\WishlistGraphQl\Model\FileSupport\BuyRequest;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+use Magento\Wishlist\Model\Wishlist\BuyRequest\BuyRequestDataProviderInterface;
+use Magento\Wishlist\Model\Wishlist\Data\WishlistItem;
+
+/**
+ * Data provider for custom options buy requests
+ */
+class CustomizableOptionDataProvider implements BuyRequestDataProviderInterface
+{
+    private const PROVIDER_OPTION_TYPE = 'custom-option';
```

Comment:
1. Please don't use `private` modifiers, since this prevent's us from accessing `private` properties and methods in the extended classes. Use `protected` instead.
2. Please make constants public.